integration/specs/CodeInput/codeInput-1.spec.js

```diff
@@ -0,0 +1,107 @@
+        browser.keys(['', '2']);
+        expect(codeInput.getFocusedIndex()).toBe(1);
+    });
+    it('should remove value from first input when first input is selected, has a value and "Backspace" key is pressed', () => {
+        const codeInput = new PageCodeInput(CODEINPUT);
+        browser.keys(['', '2']);
+        expect(codeInput.getInputValueByIndex(0)).toBe('2');
+        browser.keys(DELETE_KEY);
+        expect(codeInput.getInputValueByIndex(0)).toBe('');
+    });
+    it('should return focus to the first input and clear his value when the first input has a value, the second input is empty, is focused and "Backspace" key is pressed', () => {
+        const codeInput = new PageCodeInput(CODEINPUT);
+        browser.keys(['', '2']);
+        expect(codeInput.getInputValueByIndex(0)).toBe('2');
+        expect(codeInput.getFocusedIndex()).toBe(1);
+        browser.keys(DELETE_KEY);
+        expect(codeInput.getInputValueByIndex(0)).toBe('');
+    });
+    it('should keep focus on the first input when input is empty and "Backspace" key is pressed', () => {
+        const codeInput = new PageCodeInput(CODEINPUT);
+        expect(codeInput.getFocusedIndex()).toBe(0);
+        expect(codeInput.getInputValueByIndex(0)).toBe('');
+        browser.keys(DELETE_KEY);
+        expect(codeInput.getFocusedIndex()).toBe(0);
+    });
+    it('should keep focus on the last input when the last input value is filled with a number', () => {
+        const codeInput = new PageCodeInput(CODEINPUT);
+        browser.keys(['', '2']);
+        browser.keys(['', '2']);
+        browser.keys(['', '2']);
+        browser.keys(['', '2']);
+        expect(codeInput.getFocusedIndex()).toBe(3);
+    });
+    it('should keep focus in the current input when we type a letter', () => {
+        const codeInput = new PageCodeInput(CODEINPUT);
+        expect(codeInput.getFocusedIndex()).toBe(0);
+        browser.keys(['', 'A']);
+        expect(codeInput.getFocusedIndex()).toBe(0);
+    });
+    it('should keep current input focus when unfocused inputs are clicked', () => {
+        const codeInput = new PageCodeInput(CODEINPUT);
+        expect(codeInput.getFocusedIndex()).toBe(0);
+        codeInput.clickInputByIndex(2);
+        expect(codeInput.getFocusedIndex()).toBe(0);
+    });
+    it('should keep focus on the second input when any input is clicked and the first input has a number already', () => {
+        const codeInput = new PageCodeInput(CODEINPUT);
+        browser.keys(['', '2']);
+        expect(codeInput.getFocusedIndex()).toBe(1);
+        codeInput.clickInputByIndex(0);
+        expect(codeInput.getFocusedIndex()).toBe(1);
+    });
+    it('should leave focus on all inputs when "Tab" key is pressed', () => {
+        const codeInput = new PageCodeInput(CODEINPUT);
+        expect(codeInput.getFocusedIndex()).toBe(0);
+        browser.keys(TAB_KEY);
+        expect(codeInput.getFocusedIndex()).toBe(-1);
```

Comment:
let's use undefined to indicate that no input is focused

Reply:
I have 3 methods returning -1, 

should I apply undefined return value on all of them or just for gestFocusedIndex ?